### ISSUE TITLE
Pod didn't start on Kubelet.

### ISSUE THREAD

**Issue by wojtek-t** [kubernetes/kubernetes#36502]
This run of scalability test:
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/kubernetes-e2e-gce-scalability/11517/
failed, because one pod never started.

Looking into kubelet log, I see:
```
I1020 09:45:05.450111    3643 kubelet.go:1592] SyncLoop (ADD, "api"): "load-medium-rc-11-4qcqh_e2e-tests-load-30-nodepods-1-t608t(e14812a5-96a9-11e6-b89e-42010af00002)"
I1020 09:45:05.751174    3643 docker_manager.go:1941] Need to restart pod infra container for "load-medium-rc-11-4qcqh_e2e-tests-load-30-nodepods-1-t608t(e14812a5-96a9-11e6-b89e-42010af00002)" because it is not found
E1020 09:47:05.837740    3643 docker_manager.go:2154] Failed to create pod infra container: RunContainerError; Skipping pod "load-medium-rc-11-4qcqh_e2e-tests-load-30-nodepods-1-t608t(e14812a5-96a9-11e6-b89e-42010af00002)": runContainer: operation timeout: context deadline exceeded
I1020 09:47:17.088461    3643 docker_manager.go:1941] Need to restart pod infra container for "load-medium-rc-11-4qcqh_e2e-tests-load-30-nodepods-1-t608t(e14812a5-96a9-11e6-b89e-42010af00002)" because it is not found
E1020 09:49:17.165065    3643 docker_manager.go:2154] Failed to create pod infra container: RunContainerError; Skipping pod "load-medium-rc-11-4qcqh_e2e-tests-load-30-nodepods-1-t608t(e14812a5-96a9-11e6-b89e-42010af00002)": runContainer: operation timeout: context deadline exceeded
E1020 09:49:17.171547    3643 generic.go:238] PLEG: Ignoring events for pod load-medium-rc-11-4qcqh/e2e-tests-load-30-nodepods-1-t608t: operation timeout: context deadline exceeded
I1020 09:49:30.088745    3643 docker_manager.go:1941] Need to restart pod infra container for "load-medium-rc-11-4qcqh_e2e-tests-load-30-nodepods-1-t608t(e14812a5-96a9-11e6-b89e-42010af00002)" because it is not found
E1020 09:51:30.160984    3643 docker_manager.go:2154] Failed to create pod infra container: RunContainerError; Skipping pod "load-medium-rc-11-4qcqh_e2e-tests-load-30-nodepods-1-t608t(e14812a5-96a9-11e6-b89e-42010af00002)": runContainer: operation timeout: context deadline exceeded
I1020 09:51:45.088798    3643 docker_manager.go:1941] Need to restart pod infra container for "load-medium-rc-11-4qcqh_e2e-tests-load-30-nodepods-1-t608t(e14812a5-96a9-11e6-b89e-42010af00002)" because it is not found
E1020 09:53:45.173738    3643 docker_manager.go:2154] Failed to create pod infra container: RunContainerError; Skipping pod "load-medium-rc-11-4qcqh_e2e-tests-load-30-nodepods-1-t608t(e14812a5-96a9-11e6-b89e-42010af00002)": runContainer: operation timeout: context deadline exceeded
I1020 09:53:57.088638    3643 docker_manager.go:1941] Need to restart pod infra container for "load-medium-rc-11-4qcqh_e2e-tests-load-30-nodepods-1-t608t(e14812a5-96a9-11e6-b89e-42010af00002)" because it is not found
E1020 09:55:57.173622    3643 docker_manager.go:2154] Failed to create pod infra container: RunContainerError; Skipping pod "load-medium-rc-11-4qcqh_e2e-tests-load-30-nodepods-1-t608t(e14812a5-96a9-11e6-b89e-42010af00002)": runContainer: operation timeout: context deadline exceeded
```

The logs come from here:
http://gcsweb.k8s.io/gcs/kubernetes-jenkins/logs/kubernetes-e2e-gce-scalability/11517/artifacts/e2e-scalability-minion-group-lzyk/

@kubernetes/goog-node @yujuhong @Random-Liu

**Comment by yujuhong** [comment]
Relevant messages from docker.log:
```
time="2016-10-20T09:47:05Z" level=error msg="containerd: start container" error="containerd: container did not start before the specified timeout" id=21fc9d8113a8892bc918397e999a7003c9963f90ce783365bde94c2e99f3f328 
time="2016-10-20T09:47:05.901414412Z" level=debug msg="Failed to unmount 172d1ba2b0ddc61e4926579446c4fa663f5f784e2a54643215cece374537a256 aufs: device or resource busy" 
time="2016-10-20T09:47:05.901492593Z" level=error msg="Error unmounting container 21fc9d8113a8892bc918397e999a7003c9963f90ce783365bde94c2e99f3f328: device or resource busy" 
time="2016-10-20T09:47:05.901516347Z" level=debug msg="GetMountID id: 21fc9d8113a8892bc918397e999a7003c9963f90ce783365bde94c2e99f3f328 -> mountID: 172d1ba2b0ddc61e4926579446c4fa663f5f784e2a54643215cece374537a256" 
time="2016-10-20T09:47:05.901536247Z" level=debug msg="Cleaning up old mountid 172d1ba2b0ddc61e4926579446c4fa663f5f784e2a54643215cece374537a256: start." 
```

The "container start" failures could just be https://github.com/docker/docker/issues/22226 where containerd operation times out when there is too much load. The issue was resolved in docker 1.12 by increasing the containerd timeout. Maybe we patch docker on GCI for this /cc @dchen1107 

As for the unmount failure, I m not sure if it's just the side-effect of the initial start failure, or a more serious issue.

**Comment by dchen1107** [comment]
I am closing this issue because: 

1) It is known docker issue in docker 1.11.2 docker/docker#22226 and the fix was in docker 1.12.x
2) If is pretty rare even with docker 1.11.2
